cli/src/regtest.rs

```diff
@@ -38,98 +41,74 @@ fn git_selfcheck() {
     }
 }
 
-fn get_top_level_dir() {
-    //TODO
+///  Simple helper to succinctly reference to the project root dir.
+use std::path::{Path, PathBuf};
+fn get_top_level_dir() -> PathBuf {
+    let revparse_raw = Command::new("git")
+        .args(["rev-parse", "--show-toplevel"])
+        .output()
+        .expect("problem invoking git rev-parse");
+    Path::new(
+        std::str::from_utf8(&revparse_raw.stdout)
+            .expect("revparse error")
+            .trim(),
+    )
+    .to_path_buf()
+}
+fn get_regtest_dir() -> PathBuf {
+    get_top_level_dir().join("regtest")
 }
 
 pub(crate) fn launch() {
-    use std::ffi::OsString;
     use std::fs::File;
     use std::io::Read;
-    use std::path::PathBuf;
-    use std::process::{Command, Stdio};
+    use std::process::Stdio;
     use std::{thread, time};
 
     //check for git itself and that we are working within a zingolib repo
     git_selfcheck();
 
-    let _ = get_top_level_dir();
-    // confirm we are in a git worktree
-
-    // get the top level directory for this repo worktree
-    let revparse_raw = Command::new("git")
-        .args(["rev-parse", "--show-toplevel"])
-        .output()
-        .expect("problem invoking git rev-parse");
-
-    let revparse = std::str::from_utf8(&revparse_raw.stdout).expect("revparse error");
-
-    // Cross-platform OsString
-    let mut worktree_home = OsString::new();
-    worktree_home.push(revparse.trim_end());
-
-    let bin_location: PathBuf = [
-        worktree_home.clone(),
-        OsString::from("regtest"),
-        OsString::from("bin"),
-    ]
-    .iter()
-    .collect();
+    let confs_dir = get_regtest_dir().join("conf");
+    let bin_location = get_regtest_dir().join("bin");
+    let logs = get_regtest_dir().join("logs");
+    let datadir = get_regtest_dir().join("data");
```

Comment:
changing this means we would have to update the README to ask the cli to point `--data-dir` to this location